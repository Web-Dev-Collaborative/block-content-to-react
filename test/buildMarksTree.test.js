const buildMarksTree = require('../src/buildMarksTree')

test('builds empty tree on empty block', () => {
  const {input} = require('./fixtures/001-empty-block')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds simple one-node tree on single, markless span', () => {
  const {input} = require('./fixtures/002-single-span')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds simple multi-node tree on markless spans', () => {
  const {input} = require('./fixtures/003-multiple-spans')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds annotated span on simple mark', () => {
  const {input} = require('./fixtures/004-basic-mark-single-span')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds annotated, joined span on adjacent, equal marks', () => {
  const {input} = require('./fixtures/005-basic-mark-multiple-adjacent-spans')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds annotated, nested spans in tree format', () => {
  const {input} = require('./fixtures/006-basic-mark-nested-marks')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds annotated spans with expanded marks on object-style marks', () => {
  const {input} = require('./fixtures/007-link-mark-def')
  expect(buildMarksTree(input)).toMatchSnapshot()
})

test('builds correct structure from advanced, nested mark structure', () => {
  const {input} = require('./fixtures/009-messy-link-text')
  expect(buildMarksTree(input)).toMatchSnapshot()
})
