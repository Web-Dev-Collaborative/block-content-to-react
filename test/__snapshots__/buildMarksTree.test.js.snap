// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds annotated span on simple mark 1`] = `
Array [
  Object {
    "_key": "cZUQGmh4",
    "_type": "span",
    "children": Array [
      "sanity",
    ],
    "mark": "code",
    "markKey": "code",
  },
  " is the name of the CLI tool.",
]
`;

exports[`builds annotated spans with expanded marks on object-style marks 1`] = `
Array [
  "A word of warning; ",
  Object {
    "_key": "toaiCqIK",
    "_type": "span",
    "children": Array [
      "Sanity",
    ],
    "mark": Object {
      "_key": "someLinkId",
      "_type": "link",
      "href": "https://sanity.io/",
    },
    "markKey": "someLinkId",
  },
  " is addictive.",
]
`;

exports[`builds annotated, joined span on adjacent, equal marks 1`] = `
Array [
  Object {
    "_key": "cZUQGmh4",
    "_type": "span",
    "children": Array [
      "A word of",
      " warning;",
    ],
    "mark": "strong",
    "markKey": "strong",
  },
  " Sanity is addictive.",
]
`;

exports[`builds annotated, nested spans in tree format 1`] = `
Array [
  Object {
    "_key": "cZUQGmh4",
    "_type": "span",
    "children": Array [
      "A word of ",
      Object {
        "_key": "toaiCqIK",
        "_type": "span",
        "children": Array [
          "warning;",
        ],
        "mark": "em",
        "markKey": "em",
      },
    ],
    "mark": "strong",
    "markKey": "strong",
  },
  " Sanity is addictive.",
]
`;

exports[`builds correct structure from advanced, nested mark structure 1`] = `
Array [
  Object {
    "_key": "a1ph4",
    "_type": "span",
    "children": Array [
      "Sanity",
    ],
    "mark": Object {
      "_key": "zomgLink",
      "_type": "link",
      "href": "https://sanity.io/",
    },
    "markKey": "zomgLink",
  },
  " can be used to power almost any ",
  Object {
    "_key": "ch4r1i3",
    "_type": "span",
    "children": Array [
      Object {
        "_key": "ch4r1i3",
        "_type": "span",
        "children": Array [
          Object {
            "_key": "ch4r1i3",
            "_type": "span",
            "children": Array [
              "app",
            ],
            "mark": "strong",
            "markKey": "strong",
          },
          " or website",
        ],
        "mark": "em",
        "markKey": "em",
      },
    ],
    "mark": Object {
      "_key": "zomgLink",
      "_type": "link",
      "href": "https://sanity.io/",
    },
    "markKey": "zomgLink",
  },
  ".",
]
`;

exports[`builds empty tree on empty block 1`] = `Array []`;

exports[`builds simple multi-node tree on markless spans 1`] = `
Array [
  "Span number one. ",
  "And span number two.",
]
`;

exports[`builds simple one-node tree on single, markless span 1`] = `
Array [
  "Plain text.",
]
`;
